deps/libgit2.mk

```diff
@@ -90,9 +72,14 @@ $(LIBGIT2_SRC_PATH)/libgit2-ssh-loop.patch-applied: $(LIBGIT2_SRC_PATH)/source-e
 		patch -p1 -f < $(SRCDIR)/patches/libgit2-ssh-loop.patch
 	echo 1 > $@
 
-$(build_datarootdir)/julia/cert.pem: $(CERTFILE)
+$(build_datarootdir)/julia/cert.pem:
+	$(eval TMP_CACERT_SHA256:=$(shell mktemp cacert-2017-09-20.XXXX.pem.sha256))
+	echo "435ac8e816f5c10eaaf228d618445811c16a5e842e461cb087642b6265a36856 *cacert-2017-09-20.pem" > $(TMP_CACERT_SHA256)
+	$(JLDOWNLOAD) $(shell pwd)/cacert-2017-09-20.pem https://curl.haxx.se/ca/cacert-2017-09-20.pem
+	sha256sum --check $(TMP_CACERT_SHA256)
```

Comment:
Unfortunately, I don't think we can just assume that `sha256sum` is available, sometimes we need to use `shasum -a 256`.   See the [`jlchecksum` tool](https://github.com/JuliaLang/julia/blob/master/deps/tools/jlchecksum) for what we've found to work reliably.  In fact, it might be neat if we could just use `jlchecksum` to do the checking here.  To do so, download the `cacert-2017-09-20.pem` file, then run `./deps/tools/jlchecksum cacert-2017-09-20.pem`.  It will say something like `"checksums not found, autogenerating..."`, then there will be two new files within `deps/checksums`.  Then, if you ask `jlchecksum` to verify a file with the same filename, it will check against those checksums within the build tree.

Reply:
I have modified this section to use $(JLCHECKSUM), but I feel that the MD5 checksum verification as a fallback is not recommended for security ([similar to using SHA1 checksums](https://blog.mozilla.org/security/2014/09/23/phasing-out-certificates-with-sha-1-based-signature-algorithms/)).

Since there is only SHA512 and MD5 checksum verification with the jlchecksum script, perhaps the build should fail if the system does not provide more than MD5 checksum verifications.